wagtail_localize_rws_languagecloud/models.py

```diff
@@ -111,11 +117,41 @@ def is_failed(self):
         return self.lc_source_file_id == "" and self.create_attempts >= 3
 
     @property
-    def instance_is_published(self):
+    def published_status(self):
+        PUBLISHED = gettext_lazy("Translations published")
+        RECEIVED = gettext_lazy("Translations received")
+        READY_TO_REVIEW = gettext_lazy("Translations ready for review")
+
         instance = self.translation.get_target_instance()
+
         if not isinstance(instance, Page):
-            return True
-        return not instance.has_unpublished_changes
+            # snippets are always published immediately
+            return PUBLISHED
+
+        if not self.revision:
+            """
+            There's a couple of reasons we can end up here:
+            1. For legacy reasons: we didn't always save a revision on this object.
+               Objects created with <=0.8.0 willl have a NULL revision
+            2. If we threw an error trying to call save_target() in import_po()
+
+            In this case, say we've got the translations back
+            but we don't know if they are published or not
+            """
+            return RECEIVED
+
+        if (
+            self.revision == instance.get_latest_revision()
+            and not instance.has_unpublished_changes
+        ):
+            return PUBLISHED
+
+        if (self.revision != instance.get_latest_revision()) and (
+            self.revision.created_at < instance.get_latest_revision().created_at
```

Comment:
question: could there ever be a case where the created_at for the revision recorded against the file would not be <= instance.get_latest_revision()?

Reply:
Essentially the condition I am/was trying to express here is "a more recent revision of `instance` than `self.revision` exists".

However, reading this again with fresh eyes it occurs to me there's actually 2 distinct cases to consider here:

- One or more revisions of `instance` which are more recent than `self.revision` exist and one or more of those revisions is published
- One or more revisions of `instance` which are more recent than `self.revision` exist and none of those revisions are published

I think in the first case we label that as "Translations published"
For the second one I wonder if that is another case of "Translations ready for review" or another different (but somewhat difficult to summarise) status?

Reply:
May I suggest you pose that question to Lucas and the team?

Reply:
Have posted it on slack. I'll see where that goes then update this.